File: src/main/java/edu/marmara/dto/TranscriptGetDTO.java
package edu.marmara.dto;

import java.util.List;

public class TranscriptGetDTO {
    private Double gpa;
    private Integer passedCredit;
    private Integer failedCredit;
    private List<String> passedCourses;
    private List<String> failedCourses;
    private List<String> notTakenCourses;
    private List<String> currentlyTakenCourses;

    public TranscriptGetDTO() {
    }

    public TranscriptGetDTO(Double gpa, Integer passedCredit, Integer failedCredit, List<String> passedCourses, List<String> failedCourses, List<String> notTakenCourses, List<String> currentlyTakenCourses) {
        this.gpa = gpa;
        this.passedCredit = passedCredit;
        this.failedCredit = failedCredit;
        this.passedCourses = passedCourses;
        this.failedCourses = failedCourses;
        this.notTakenCourses = notTakenCourses;
        this.currentlyTakenCourses = currentlyTakenCourses;
    }

    public Double getGpa() {
        return gpa;
    }

    public void setGpa(Double gpa) {
        this.gpa = gpa;
    }

    public Integer getPassedCredit() {
        return passedCredit;
    }

    public void setPassedCredit(Integer passedCredit) {
        this.passedCredit = passedCredit;
    }

    public Integer getFailedCredit() {
        return failedCredit;
    }

    public void setFailedCredit(Integer failedCredit) {
        this.failedCredit = failedCredit;
    }

    public List<String> getPassedCourses() {
        return passedCourses;
    }

    public void setPassedCourses(List<String> passedCourses) {
        this.passedCourses = passedCourses;
    }

    public List<String> getFailedCourses() {
        return failedCourses;
    }

    public void setFailedCourses(List<String> failedCourses) {
        this.failedCourses = failedCourses;
    }

    public List<String> getNotTakenCourses() {
        return notTakenCourses;
    }

    public void setNotTakenCourses(List<String> notTakenCourses) {
        this.notTakenCourses = notTakenCourses;
    }

    public List<String> getCurrentlyTakenCourses() {
        return currentlyTakenCourses;
    }

    public void setCurrentlyTakenCourses(List<String> currentlyTakenCourses) {
        this.currentlyTakenCourses = currentlyTakenCourses;
    }
}

File: src/python/src/main/marmara/dto/TranscriptGetDTo.py

class TranscriptGetDTO:
    def __init__(self, gpa=None, passed_credit=None, failed_credit=None, passed_courses=None, failed_courses=None, not_taken_courses=None, currently_taken_courses=None):
        self.gpa = gpa
        self.passed_credit = passed_credit
        self.failed_credit = failed_credit
        self.passed_courses = passed_courses or []
        self.failed_courses = failed_courses or []
        self.not_taken_courses = not_taken_courses or []
        self.currently_taken_courses = currently_taken_courses or []

    def add_currently_taken_course(self, course):
        self.currently_taken_courses.append(course)

    def add_failed_course(self, course):
        self.failed_courses.append(course)

    def add_not_taken_course(self, course):
        self.not_taken_courses.append(course)

    def add_passed_course(self, course):
        self.passed_courses.append(course)

    @property
    def gpa(self):
        return self._gpa

    @gpa.setter
    def gpa(self, value):
        self._gpa = value or 0.0

    @property
    def passed_credit(self):
        return self._passed_credit

    @passed_credit.setter
    def passed_credit(self, value):
        self._passed_credit = value or 0

    @property
    def failed_credit(self):
        return self._failed_credit

    @failed_credit.setter
    def failed_credit(self, value):
        self._failed_credit = value or 0
